sql/moz-fx-data-shared-prod/telemetry_derived/main_summary_v4/part1.sql

```diff
@@ -45,6 +45,25 @@ CREATE TEMP FUNCTION extract_popup_notification_stats(h STRING) AS (
   )
 );
 
+WITH extracted_histograms AS (
+  SELECT
+    document_id,
+    client_id,
+    sample_id,
+    mozfun.hist.extract(payload.histograms.ssl_handshake_result) AS ssl_handshake_result,
+    mozfun.hist.extract(
+      payload.histograms.plugins_notification_shown
+    ) AS plugins_notification_shown,
+    mozfun.hist.extract(
+      payload.histograms.plugins_notification_user_action
+    ) AS plugins_notification_user_action
+  FROM
+    `moz-fx-data-shared-prod.telemetry.main`
+  WHERE
+    DATE(submission_timestamp) = @submission_timestamp
+    AND normalized_app_name = 'Firefox'
+    AND document_id IS NOT NULL
+)
```

Comment:
Would this address the complexity concern?

Reply:
See comment above. We should avoid adding a join to this query, which would likely be a significant drag on performance.